src/dotnet/CarbonAware.Plugins.JsonReaderPlugin/CarbonAwareJsonReaderPlugin.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System.Collections;
+using CarbonAware.Model;
+using Microsoft.Extensions.Logging;
+using System.Reflection;
+using Newtonsoft.Json;
+
+namespace CarbonAware.Plugins.JsonReaderPlugin;
+
+public class CarbonAwareJsonReaderPlugin : ICarbonAware
+{
+    private readonly ILogger<CarbonAwareJsonReaderPlugin> _logger;
+
+
+    public CarbonAwareJsonReaderPlugin(ILogger<CarbonAwareJsonReaderPlugin> logger)
+    {
+        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+    }
+
+    public async Task<IEnumerable<EmissionsData>> GetEmissionsDataAsync(IDictionary props)
+    {
+        var data = GetSampleJson();
+
+        return await Task.Run(() => GetFilteredData(data, props));
+    }
+
+    private IEnumerable<EmissionsData> GetFilteredData(IEnumerable<EmissionsData> data, IDictionary props) {
+        var l = props[CarbonAwareConstants.Locations] as IEnumerable<string>;
+        List<String> locations = l !=null ? l.ToList() : new List<string>();
+
+        var s = props[CarbonAwareConstants.Start];
+        var start = DateTime.Now;
+        if (s != null)
+        {
+            DateTime.TryParse(s.ToString(), out start);
+        }
+        var e = props[CarbonAwareConstants.End];
```

Comment:
It may improve readability to separate this logic out into methods like `private DateTime GetEndDateFromProps(IDictionary props)`  here and for each expected property.